app/jobs/unzip_job.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'zip'
+
+# Change work version from a zipfile type to a browser type by unzipping the attached zip file.
+class UnzipJob < BaseDepositJob
+  queue_as :default
+
+  def perform(work_version)
+    zip_attached_file = work_version.attached_files.last
+    destroy_attached_files_except(zip_attached_file, work_version)
+    Zip::File.open(filepath_for(zip_attached_file)) do |zip_file|
+      Dir.mktmpdir do |temp_dir|
+        zip_file.each do |entry|
+          next if ignore?(entry)
+
+          attach_file(entry:, temp_dir:, work_version:)
+        end
+      end
+    end
+    zip_attached_file.destroy
+    work_version.upload_type = 'browser'
+    work_version.unzip_complete!
+  end
+
+  private
+
+  def destroy_attached_files_except(except_attached_file, work_version)
+    work_version.attached_files.each do |attached_file|
+      work_version.attached_files.destroy(attached_file) unless attached_file == except_attached_file
+    end
```

Comment:
There's likely a more database-/AR-optimized way to do this but we're invoking this in a job so 🤷🏻‍♂️ ?

Reply:
When I was writing this code I shared that feeling, but couldn't quite put my finger on the code.

Reply:
How about something like...

```ruby
work_version.attached_files.excluding(except_attached_file).destroy_all
```

Reply:
Why couldn't I figure that out?

Reply:
It took a bit of digging once I convinced myself it should be possible, FWIW!